test/test_unary_ufuncs.py

```diff
@@ -1179,6 +1179,38 @@ def test_i0_special(self, device, dtype):
         t = torch.tensor([inf, -inf, nan], device=device, dtype=dtype)
         self.assertTrue(torch.i0(t).isnan().all())
 
+    @dtypesIfCUDA(*torch.testing.get_all_fp_dtypes())
+    @dtypes(torch.bfloat16, torch.float32, torch.float64)
+    @unittest.skipIf(not TEST_SCIPY, "SciPy not found")
+    def test_special_i0e_vs_scipy(self, device, dtype):
+        def check_equal(t):
+            # Test by comparing to scipy
+            actual = torch.special.i0e(t)
+            if dtype is torch.bfloat16:
+                t = t.to(torch.float32)
+            expected = scipy.special.i0e(t.cpu().numpy())
+
+            # Casting down for dtype float16 is required since scipy upcasts to float32
+            if dtype is torch.bfloat16 or dtype is torch.float16:
```

Comment:
How interesting is supporting half and bfloat16 for these special operations, do you think?

Should we just support them in float and double only for now?

Reply:
My thought was that `i0` and `i0e` should atleast support same dtypes (note: i0e supports int -> float promotion while i0 doesn't as of now)

Reply:
I agree, but we could change what types i0 supports, too. I'm just curious what your thinking on which dtypes are important is, and if we should start with float and double and see if users request lower precision floating types?